shared_model/validators/proposal_validator.hpp

```diff
@@ -64,7 +64,9 @@ namespace shared_model {
         for (const auto &tx : prop->transactions()) {
           validateTransaction(reason, tx);
         }
-        answer.addReason(std::move(reason));
+        if (not reason.second.empty()) {
```

Comment:
Because it does not work otherwise. If answer has even empty reason, builder will see that answer contains errors and return error result. Because of that, user may see exception which has empty error message, whereas there should not be any errors at all